tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -237,6 +237,7 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
 
   private def checkMethodDef(methodDef: MethodDef,
       classDef: LinkedClass): Unit = withPerMethodState {
+    // scalastyle:off return
```

Comment:
Consider setting this for the entire IR checker once (maybe even adding a comment justifying why). As in:

```scala
// scalastyle:off return
// In the IR-Checker we allow returns for early returns for improved readability.
```